Tests/Functional/Asynchronous/main.swift

```diff
@@ -0,0 +1,184 @@
+// RUN: %{swiftc} %s -o %{built_tests_dir}/Asynchronous
```

Comment:
@briancroom Thank you for the excellent feedback. You're absolutely right, and as I responded to feedback on this pull request I definitely found the tests extremely hard to update.

- First off, your suggestion to place the `// CHECK:` lines close to what they are testing is spot-on. I've done so, and the tests are much easier to read.
- Unfortunately, the line needs to begin with `// CHECK:`, meaning it can't be indented. I'll fix this in a separate pull request, since I think it's nice to have identical indentation level.
- I also split up the `Asynchronous/main.swift` test into three separate files one for each test case. This reduces the reasons for line numbers to change--those are annoying to update.

It'd also be really nice to "name" line numbers. Here's an example:

```swift
// By storing the line number in a named variable, even if the line
// upon which the failure is reported in the source file changes, the
// test won't have to be updated.
func testBar() {
    // CHECK: .*/Tests/Functional/Foo/main.swift:[[LINE_NUMBER]]: error: FooCase.testBar : .*
    XCTAssert(false) // LINE_NUMBER: #line
}
```

We could also do the following:

```swift
// The check line references the line number directly.
XCTAssert(false) // CHECK: .*/Tests/Functional/Foo/main.swift:[[#line]]: error: FooCase.testBar : .*
```

Your suggestions gave me several ideas on how we can make the functional tests more maintainable--thanks!! :heart_eyes: